api/app/tests/medium/routers/test_tags.py

```diff
@@ -121,6 +121,27 @@ def test_get_tags():
     assert data[0]["tag_id"] == str(tag3.parent_id)
 
 
+def test_get_tag():
+    create_user(USER1)
+
+    str1 = "a1:a2:"
+    tag1 = create_tag(USER1, str1)
+    str2 = "b1:b2:b3"
+    tag2 = create_tag(USER1, str2)
+
+    data = assert_200(client.get("/tags/" + str(tag1.tag_id), headers=headers(USER1)))
+    assert data["tag_id"] == str(tag1.tag_id)
+    assert data["tag_name"] == str(tag1.tag_name)
+    assert data["parent_id"] == str(tag1.parent_id)
+    assert data["parent_name"] == str(tag1.parent_name)
+
+    data = assert_200(client.get("/tags/" + str(tag2.tag_id), headers=headers(USER1)))
+    assert data["tag_id"] == str(tag2.tag_id)
+    assert data["tag_name"] == str(tag2.tag_name)
+    assert data["parent_id"] == str(tag2.parent_id)
+    assert data["parent_name"] == str(tag2.parent_name)
+
+
```

Comment:
テストについて
- 複数のデータを作成+取得してますが、これはテストに二つのタグが必要というより、パラメータが変更して確かめてます。これはpytest.parametrize を使ってパラメタライズ化しましょう
- assert_200 は複数の責務 (戻ってきたデータをjson化すること、response code が200以外が帰ったら失敗すること)を確かめており、関数名からそれを察知することはできません。またこのテストの主題は HTTP レスポンスが200であることを確かめたいわけではないはずなので、レスポンスコードのに対するassertは不要かと


Given, When, Then を念頭に書くと以下のようになるかと思います
```suggestion
@pytest.mark.parametrize(
    "tag_name",
    [
        "a1:a2:",
        "b1:b2:b3"
    ],
)
def test_it_should_return_tag_data_when_it_requested_existing_tagid(tag_name):
    # Given
    # 予めtagが作成されており
    create_user(USER1)
    tag = create_tag(USER1, tag_name)

    # When
    # ユーザがtag_id を指定してリクエストした場合
    response = client.get("/tags/" + str(tag.tag_id), headers=headers(USER1)))
    fetched_tag = response.json()
    
    # Then
    # 取得したデータには、タグ作成時と同じ情報が含まれている
    assert fetched_tag["tag_id"] == str(tag.tag_id)
    assert fetched_tag["tag_name"] == str(tag1.tag_name)
    assert fetched_tag["parent_id"] == str(tag.parent_id)
    assert fetched_tag["parent_name"] == str(tag.parent_name)
```

Reply:
指摘の通り修正しました。